kubernetes/resource_kubernetes_job_test.go

```diff
@@ -56,10 +76,33 @@ func TestAccKubernetesJob_basic(t *testing.T) {
 					resource.TestCheckResourceAttr("kubernetes_job.test", "spec.0.completions", "1"),
 					resource.TestCheckResourceAttr("kubernetes_job.test", "spec.0.parallelism", "1"),
 					resource.TestCheckResourceAttr("kubernetes_job.test", "spec.0.manual_selector", "true"),
+					resource.TestCheckResourceAttr("kubernetes_job.test", "spec.0.selector.0.match_labels.foo", "bar"),
 					resource.TestCheckResourceAttr("kubernetes_job.test", "spec.0.template.0.spec.0.container.0.name", "hello"),
 					resource.TestCheckResourceAttr("kubernetes_job.test", "spec.0.template.0.spec.0.container.0.image", "alpine"),
 				),
 			},
+			{
+				Config: testAccKubernetesJobConfig_recreated_image(name),
```

Comment:
@alexsomesan do you know of a way to ensure a resource is updated or recreated during acceptance test steps checks?

In the context of kubernetes, this could probably be achieved comparing resource `uid` between steps, but it is not trivial, and it could most certainly be achieved more generically at the terraform `TestStep` level.

Reply:
There isn't any canonical way. For Kubernetes specifically I would look at `metadata.generation` or `metadata.creation_timestamp` and track those between the different test steps. I'm not entirely sure how `generation` behaves between updates, but the creation timestamp should not change.

Reply:
I'll try this. In the mean time, I found https://github.com/hashicorp/terraform-plugin-sdk/issues/77

Reply:
I've submitted https://github.com/hashicorp/terraform-plugin-sdk/pull/329, based on https://github.com/hashicorp/terraform-plugin-sdk/issues/222

Reply:
@pdecat you were right about looking at the UID, btw. I've been adding a bunch of checks to ensure resources aren't deleted and re-created. I built off of someone else's work who implemented the [ForceNew checks](https://github.com/hashicorp/terraform-provider-kubernetes/blob/f7d2f8d3ccd25b8baddc262351d749a24bae42b3/kubernetes/resource_kubernetes_persistent_volume_test.go#L900-L913) in the provider tests. So that's how the UID comparison is done.